services-core/src/gas_price.rs

```diff
@@ -15,12 +16,12 @@ pub trait GasPriceEstimating {
 }
 
 /// Creates the default gas price estimator for the given network.
-pub fn create_estimator(
-    network_id: u64,
+pub async fn create_estimator(
     http_factory: &HttpFactory,
     web3: &Web3,
 ) -> Result<Arc<dyn GasPriceEstimating + Send + Sync>> {
-    Ok(match gas_station::api_url_from_network_id(network_id) {
+    let chain_id = web3.eth().chain_id().compat().await?.as_u64();
```

Comment:
ubernit:
```suggestion
    let network_id = web3.net().version().compat().await?;
```

I think we want to use the network ID here, and not the chain ID (which is for signing). This is really nit-picky, and my main argument here is that the version of Ganache we are using only allows specifying the network ID and not the chain ID (which is hard coded to `1337` if I am not mistaken).

```
$ docker run -d -p 8545:8545 docker.io/trufflesuite/ganache-cli:v6.7.0 -i 5777
$ printf "%d\n" $(curl -s "http://127.0.0.1:8545" -d '{"jsonrpc":"2.0","method":"eth_chainId","params":[]}' | jq -r '.result')
1337
```

Reply:
I see, I got confused by the rust web3 documentation. It says "Returns protocol version" when it actually returns the network id. It also returns a string for some reason instead of a number.
I don't think it matters with the current gas station selection anyway because we only need to differentiate mainnet and rinkeby but network id is more appropriate. Maybe use chain id now and switch to network id after making a PR to rust web3 to update the method.

Reply:
> It also returns a string for some reason instead of a number

Yes, this is confusing. Its because the official Ethereum JSON RPC spec says that it returns a string, not a number: https://eth.wiki/json-rpc/API#net_version